cache/sharded_cache.h

```diff
@@ -9,29 +9,36 @@
 
```

Comment:
The main change here is the 2 protected functions that call the Hasher and the ShardExtractor.